catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/FrameCenterUpdateField.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.video.stream.mpegts.plugins;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.alliance.libs.klv.AttributeNameConstants;
+import org.codice.alliance.libs.klv.GeometryOperator;
+import org.codice.alliance.libs.klv.GeometryOperatorList;
+import org.codice.alliance.libs.klv.GeometryUtility;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.io.WKTReader;
+import com.vividsolutions.jts.io.WKTWriter;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+/**
+ * Update the frame-center field based on the frame-center data in the children. The coordinates
+ * of the child frame-center linestrings are concatenated into a single linestring and saved to
+ * the parent.
+ * <p>
+ * This is not thread-safe.
+ */
+@NotThreadSafe
+public class FrameCenterUpdateField implements UpdateParent.UpdateField {
+
+    private static final int MAX_SIZE = 1000;
+
+    private final GeometryOperator geometryOperator;
+
+    private Geometry intermediateGeometry;
+
+    /**
+     * @param geometryOperator applied to the final linestring before it is saved to the parent
+     */
+    public FrameCenterUpdateField(GeometryOperator geometryOperator) {
+        this.geometryOperator = new GeometryOperatorList(Arrays.asList(geometryOperator,
+                new GeometrySubsample(MAX_SIZE)));
+    }
+
+    @Override
+    public void updateField(Metacard parent, List<Metacard> children) {
+
+        WKTReader wktReader = new WKTReader();
```

Comment:
We've never injected in all the other places it's used. Do you want to make an argument as to why we should inject it? Also, WKTReader is a class, not an interface.